### PR TITLE
[dart][dart-dio] Correctly type responses and futures

### PR DESCRIPTION
**Pull request by kuhnroyal**
* don't rely on implicit dynamics
* add space between return type and some method names

Note: `http.Response` is not generically typed for dart but `dio.Response<Type>` is.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [x] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

CC @swipesight (2018/09) @jaumard (2018/09) @josh-burton (2019/12) @amondnet (2019/12) @sbu-WBT (2020/12) @kuhnroyal (2020/12) @agilob (2020/12)

### CODE REVIEW

File: samples/client/petstore/dart-dio/petstore_client_lib/lib/api/pet_api.dart [hunk 1]
```diff
@@ -18,7 +18,7 @@ class PetApi {
     /// Add a new pet to the store
     ///
     /// 
-    Future<Response>addPet(
+    Future<Response<void>> addPet(
```
Comment (agilob) [inline samples/client/petstore/dart-dio/petstore_client_lib/lib/api/pet_api.dart:21]: are you sure this is correct for adding a pet? persumably it's for POST and `_dio.request` has `bodyData`.

Why is uploading file different than uploading object Pet?

Comment (kuhnroyal) [reply samples/client/petstore/dart-dio/petstore_client_lib/lib/api/pet_api.dart:21]: No response defined here: https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/resources/3_0/petstore-with-fake-endpoints-models-for-testing.yaml#L42

And 200 with ApiResponse here: https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/resources/3_0/petstore-with-fake-endpoints-models-for-testing.yaml#L259 200

Comment (kuhnroyal) [reply samples/client/petstore/dart-dio/petstore_client_lib/lib/api/pet_api.dart:21]: Btw this is the response, the `bodyData` doesn't matter.

**Review by agilob — APPROVED** [review]

**Comment by wing328** [comment]
Shippable CI failure not related to this change.

**Pull request merged**